third_party/move/move-model/src/model.rs

```diff
@@ -3376,7 +3456,24 @@ impl<'env> StructEnv<'env> {
             })
     }
 
-    /// Return the number of fields in the struct.
+    /// Get fields of a particular variant.
+    pub fn get_fields_of_variant(
+        &'env self,
+        variant: Symbol,
+    ) -> impl Iterator<Item = FieldEnv<'env>> {
+        self.data
+            .field_data
+            .values()
+            .filter(|data| data.variant == Some(variant))
```

Comment:
This may be very expensive.  Add a warning to the function comment.

Reply:
Nothing particular expensive here, iterating over a list with a few elements.

Reply:
Oh, hadn't realized that this is in `StructEnv`.  I thought it was more global.